crates/solver/src/driver/solver_settlements.rs

```diff
@@ -123,10 +123,6 @@ pub fn retain_mature_settlements(
             let mut new_order_added = false;
 
             for (index, (_, settlement)) in settlements.iter().enumerate() {
-                if valid_settlement_indices.contains(&index) {
-                    break;
-                }
```

Comment:
Thanks for finding this error. 👍 
In the backend sync you mentioned that you are worried about the performance of this algorithm. I think turning `break` into `continue` should still fix the issue you found while avoiding to hash orders unnecessarily.
Not really that important though, since you mentioned to look into an alternative implementation anyway.

Reply:
Yes, let's make it `continue`.